src/lando/main/scm/abstract_scm.py

```diff
@@ -229,3 +229,14 @@ def merge_onto(
 
         Return the SHA of the newly created merge commit.
         """
+
+    @abstractmethod
+    def tag(self, name: str, target: str | None):
+        """Create a new tag called `name` on the `target` commit.
+
+        If `target` is `None`, use the currently checked out commit.
+        """
```

Comment:
For the sake of maintaining the Pushlog, could this return a `CommitData` object for the commit pointed to by the Tag?

I think implementations could just return `self.describe_commit(name)`

Then we can do something like this in the action:

```
tagged_commit_data = scm.tag(name, target)
pushlog.add_tag(name, tagged_commit_data)
```